ikarus/utils/dirichletvalues.hh

```diff
@@ -97,11 +101,32 @@ public:
   }
 
   /**
-   * \brief Function to fix (set boolean values to true or false) of degrees of freedom
+   * \brief Fixes (set boolean value to true) a specific degree of freedom
+   *
+   * \param i An index indicating the DOF number to be fixed
+   * \param flag Boolean indicating whether the DOF should fixed or not
+   */
+  template <typename MultiIndex>
+  requires(not std::integral<MultiIndex>)
+  void setSingleDOF(const MultiIndex i, bool flag) {
+    dirichletFlagsBackend_[i] = flag;
+  }
+
+  /**
+   * \brief Fixes (set boolean value to true) a specific degree of freedom
    *
    * \param i An index indicating the DOF number to be fixed
+   * \param flag Boolean indicating whether the DOF should fixed or not
+   */
+  void setSingleDOF(std::size_t i, bool flag) { dirichletFlags_[i] = flag; }
```

Comment:
This should be not needed. Why does the old `typename Basis::MultiIndex` as argument not work? For flat basis this can be called with ({i}) to construct the dim==1 size MultiIndex for the old function signature. But anyway usually you should have a multiIndex anyway at hand

Reply:
If it does not work nicely you can also use the method above and constrain the template that you have an integral type or the same type as `typename Basis::MultiIndex`.

Reply:
So in the past we had `isConstrained(size_t)` and `isConstrained(MultiIndex)` so i thought to make the interface symmetric i will add both methods to `setSingleDOF()`. So i would either remove both `size_t` methods oder keep both

Reply:
Okay neither 
```cpp
  dirichletValues2.setSingleDOF(2, false);
  ```
  nor 
  ```cpp
  dirichletValues2.setSingleDOF({i}, false);
  ```
  compiles (on GCC) without the `size_t` function

Reply:
Also constraining the functin like that 
```cpp
  requires(std::same_as<MultiIndex, typename Basis::MultiIndex> or std::integral<MultiIndex>)
  ```
  does not work, also the function body is different, when passing a multiindex, the dirichletBackend is set, when passing a size_t the dirichletFlags is set